flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultPartition.java

```diff
@@ -80,6 +80,8 @@
 	/** Type of this partition. Defines the concrete subpartition implementation to use. */
 	private final ResultPartitionType partitionType;
 
+	private boolean isManagedExternally;
```

Comment:
For externally managed result partitions, we may also need to bypass the checking of pendingReference in createSubpartitionView and onConsumedSubpartition to support reading multiple times.

Reply:
This is a very good point, addressed in #8654.